src/Lykke.Service.KycReports.AzureRepositories/Reports/ReportsRepository.cs

```diff
@@ -0,0 +1,148 @@
+namespace Lykke.Service.KycReports.AzureRepositories.Reports
+{
+    public class ReportsRepository : IKycReportsRepository
+    {
+        private readonly INoSQLTableStorage<ReportRowEntity> _tableStorage;
+
+        public ReportsRepository(INoSQLTableStorage<ReportRowEntity> tableStorage)
+        {
+            _tableStorage = tableStorage;
+        } 
+
+        public async Task InsertRow(IKycReportDailyLeadership row)
+        {
+            var reportDataRow = ReportRowEntity.Create(KycReportType.KycReportDailyLeadership, row, row.ReportDay.Ticks.ToString());
+            await _tableStorage.InsertOrReplaceAsync(reportDataRow);
+        }
+
+        public async Task InsertRow(IKycOfficerStatsDataReport row)
+        {
+            var reportDataRow = KycOfficerStatsReportEntity.Create(row, row.RowId);
+            await _tableStorage.InsertOrReplaceAsync(reportDataRow);
+        }
+
+        public async Task InsertRow(IKycOfficersPerformanceRow row)
+        {
+            var reportDataRow = KycOfficersPerformanceReportEntity.Create(row, row.RowId);
+            await _tableStorage.InsertOrReplaceAsync(reportDataRow);
+        }
+
+        public async Task<IEnumerable<string>> GetJsonData(KycReportType reportType, bool isDescending = false, IEnumerable<string> rowKeys = null)
+        {
+            string partitionKey;
+
+            switch (reportType)
+            {
+                case KycReportType.KycReportDailyLeadership:
+                    partitionKey = ReportRowEntity.GeneratePartitionKey(KycReportType.KycReportDailyLeadership);
+                    break;
+                default:
+                    partitionKey = null;
+                    break;
+            }
+
+
+            if (!string.IsNullOrWhiteSpace(partitionKey))
+            {
+                IEnumerable<ReportRowEntity> data;
+                if (rowKeys == null)
+                    data = await _tableStorage.GetDataAsync(partitionKey);
+                else
+                    data = await _tableStorage.GetDataAsync(partitionKey, rowKeys);
+
+                var jsonRows = isDescending
+                    ? data.OrderByDescending(d => d.Timestamp).Select(d => d.JsonRow)
+                    : data.Select(d => d.JsonRow);
+                
+                //var jsonReport = $"[\r\n{string.Join(", \r\n", jsonRows)}\r\n]";
+
+                return jsonRows;
+            }
+
+            return new List<string>();
+        }
+
+        public async Task<IEnumerable<T>> GetRows<T>(IEnumerable<string> rowKeys = null)
+        {
+            var type = typeof (T);
+            string partitionKey = null;
+
+            if (type == typeof (KycReportDailyLeadership))
+                partitionKey = ReportRowEntity.GeneratePartitionKey(KycReportType.KycReportDailyLeadership);
+                
+            if (!string.IsNullOrWhiteSpace(partitionKey))
+                return await GetReportRows<T>(partitionKey, rowKeys);
+
+            return new List<T>();
```

Comment:
Why can't we init partition key for all other types of KycReportType ?

Reply:
This method was removed